Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/FareCalculator.hs

```diff
@@ -152,6 +160,7 @@ calculateFareParameters params = do
   logTagInfo "FareCalculator" $ "Initiating fare calculation for organization " +|| params.farePolicy.merchantId ||+ " and vehicle variant " +|| params.farePolicy.vehicleVariant ||+ ""
   let fp = params.farePolicy
   id <- generateGUID
+  unless (isJust (fp.averageSpeedOfVehicle) && fromJust (fp.averageSpeedOfVehicle) > 0) $ throwError AverageVehicleSpeedLessThanZero
```

Comment:
Partial functions should be avoided even though you have an isJust check. Better to use case match